observations/obs_converters/AIRS/convert_amsu_L1.f90

```diff
@@ -107,31 +116,73 @@ program convert_amsu_L1
+call initialize_amsua_netcdf(l1_files(1),verbose)
 
 ! FIXME ... maybe ... are all the files guaranteed to have the
 ! same channels - I think so.
 call channel_list_to_indices(channel_list, use_channels)
 
-! for each input file
+! Figure out how many (more) observations there may be in the output.
+! It is impossible to predict how the lat/lon subsetting may affect this
+! number, but the thinning and channels can be taken into account.
+
+max_num = max_possible_obs(filecount, cross_track_thin, along_track_thin, use_channels)
+
+! either read existing obs_seq or create a new one
+
+call append_or_create(append_output,outputfile,max_num,big_sequence)
+
+! read from netCDF file into a derived type that holds all the information
+! convert derived type information to DART sequence for that file
+! combine that sequence to the output sequence
+! release the file sequence memory in preparation for creating it new next iteration
+
 do ifile = 1,filecount
 
-   ! read from HDF file into a derived type that holds all the information
    call read_amsua_bt_netCDF_granule(l1_files(ifile), granule)   
 
-   ! convert derived type information to DART sequence
-   call make_obs_sequence(seq, granule, lon1, lon2, lat1, lat2, &
+   call make_obs_sequence(small_sequence, granule, lon1, lon2, lat1, lat2, &
                           use_channels, along_track_thin, cross_track_thin)
 
-   ! write the sequence to a disk file
-   call write_obs_seq(seq, outputfile) 
- 
-   ! release the sequence memory
-   call destroy_obs_sequence(seq)
+   num_inserted = combine_sequences(small_sequence, big_sequence, l1_files(ifile))
+
+   if (verbose > 2) then
+      call print_obs_seq_summary(small_sequence)
+   elseif (verbose > 1) then
+      write(string1,*)trim(l1_files(ifile))//' added ',num_inserted,' observations.'
+      call error_handler(E_MSG,string1,'')
+   endif
+
+   call destroy_obs_sequence(small_sequence)
 
 enddo
 
-call error_handler(E_MSG, source, 'Finished successfully.',source,revision,revdate)
+! Report on the current status of the entire observation sequence
+if (verbose > 0) then
+   call print_obs_seq_summary(big_sequence)
+endif
+
+! DISCUSSION
+! There is some discussion in the group about whether this program should
+! error out or continue if there are no observations in the sequence.
+! If used in a batch job, maybe you're OK with not having an output observation
+! sequence for this collection of files -OR- you might want to know that there
+! are no viable observations for a particular collection of files.
+! If you want to allow this program to exit without returning an error code, 
+! change E_ERR to E_MSG in the following call to error_handler().
+
+! Write the sequence to a disk file
+if ( get_num_obs(big_sequence) > 0 )  then
+   call write_obs_seq(big_sequence, outputfile) 
+else
+   call error_handler(E_ERR,'NO OBSERVATIONS TO WRITE',source, &
+              text2='see the DISCUSSION comment in the source to exit without error.')
```

Comment:
FYI: when the error_handler exits with an error in this case, the program status is 99.
When it exits with E_MSG, the program status is 0.

Reply:
yes, that's exactly what i would expect.